### DISCUSSION TITLE
Would be DefaultSettings a inheritance from Midi class settings?

### DISCUSSION THREAD

**Discussion by RobertoHE** [lathoub/Arduino-BLE-MIDI#59] [Ideas]
https://github.com/lathoub/Arduino-BLE-MIDI/blob/8dee1f7d257aa71aeb0e956632da2dd2ae9ed496/src/BLEMIDI_Settings.h#L7-L10

It would be :
` struct DefaultSettings : public MIDI_NAMESPACE::DefaultSettings`

It would allow using the same config Struct to set BleMidiTransport and MidiInterface at once. You would modify some settings of any of two layers in the same Struct. For example, edit [`MaxBufferSize`](https://github.com/lathoub/Arduino-BLE-MIDI/blob/8dee1f7d257aa71aeb0e956632da2dd2ae9ed496/src/BLEMIDI_Settings.h#L7) of BLETransport and [`HandleNullVelocityNoteOnAsNoteOff`](https://github.com/FortySevenEffects/arduino_midi_library/blob/2d64cc3c2ff85bbee654a7054e36c59694d8d8e4/src/midi_Settings.h#L62) of MidiInterface 

It may be useful for #58.

What do you think about it, @lathoub?

**Comment by lathoub** [comment]
> What do you think about it, @lathoub?

Not a fan, because the Settings in BLE are only in context of BLE (in this case to a buffer that is needed for bluetooth). Now there is a nice separation of concern.

Also (more pure technical reason): it would make the templating much more difficult (initial gut reaction).
